src/js/program/program-controller.js

```diff
@@ -44,58 +71,64 @@ class ProgramController extends Events {
         const item = model.get('playlist')[index];
 
         model.attributes.itemReady = false;
-        model.setActiveItem(index);
-        const source = getSource(item);
-        if (!source) {
-            return Promise.reject(new PlayerError(MSG_CANT_PLAY_VIDEO, ERROR_PLAYLIST_ITEM_MISSING_SOURCE));
-        }
 
         // setActiveItem above might change this.mediaController, so retrieve here.
         const { background, mediaController } = this;
 
-        // Activate the background media if it's loading the item we want to play
-        if (background.isNext(item)) {
-            // First destroy the active item so that the BGL provider can enter the foreground
-            this._destroyActiveMedia();
-            // Attach the BGL provider into the load/play chain
-            this.loadPromise = this._activateBackgroundMedia();
-            return this.loadPromise;
+        // Destroy background loading item early so we can reuse the media elements in asyncItem
+        if (!background.isNext(item)) {
+            // Loading a new item invalidates all background loading media
+            this._destroyBackgroundMedia();
         }
-        // Loading a new item invalidates all background loading media
-        this._destroyBackgroundMedia();
+        return (this.loadPromise = this.asyncItem(index, item).then((playlistItem) => {
```

Comment:
@pajong Are you saying `api.setPlaylistItem` is called by DAI while `this.asyncItem` is running?

Reply:
No, this `playlistItem` being returned from `asyncItem` is not normalized, but `this.setPlaylistItem` normalizes the same `playlistItem` and sets it - contains the same thing but a different instance